pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -542,3 +548,32 @@ func GetVMUUID() (string, error) {
 func GetUUIDFromProviderID(providerID string) string {
 	return strings.TrimPrefix(providerID, ProviderPrefix)
 }
+
+func IsUUIDSupportedNode(node *v1.Node) (bool, error) {
+	newVersion, err := version.ParseSemantic("v1.9.4")
```

Comment:
@divyenpatel @abrarshivani This change seems problematic, could we have not done this in a different way? Such as if - we can't find node using `ProviderID` then we could use `SystemUUID`? Some kubernetes distributions such as Openshift do not necessarily match kubernetes versioning schemes and hence the fix will not work for them.

Reply:
@gnufied in that case `IsUUIDSupportedNode()` is throwing error?
If that is the case, then we need to correct `func GetNodeUUID(node *v1.Node) (string, error)` function.

 

Reply:
@divyenpatel no `IsUUIDSupportedNode` will always return `true` in Kubernetes distributions which do not necessarily track kube's version naming. Basically - openshift-3.9 == kube-1.9 but after initial release openshift does not follow release pattern of kubernetes and hence openshift-3.9.40 will still report kube-1.9.0..